### ISSUE TITLE
Integers are not sorted correctly 

### ISSUE THREAD

**Issue by FilipSzm** [notepad-plus-plus/notepad-plus-plus#11023]
### Description of the Issue

Negative numbers are sorted in reverse order

### Steps to Reproduce the Issue

1. Open attached file
[numbers.txt](https://github.com/notepad-plus-plus/notepad-plus-plus/files/7841344/numbers.txt)
2. Click 'Edit' in toolbar
3. Click 'Line Operations' in Edit menu
4. Click 'Sort Lines As Integers Descending ' or 'Sort Lines As Integers Ascending ' in Line Operations submenu

### Expected Behavior

Numbers are sorted ascending or descending (depending on option in step 4)

### Actual Behavior

Non negative numbers are sorted correctly while negative numbers are sorted in reverse order (ascending for descending sort and vice versa)

### Debug Information

Notepad++ v8.1.9.3   (64-bit)
Build time : Dec  6 2021 - 19:21:37
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit) 
OS Version : 2009
OS Build : 19042.1415
Current ANSI codepage : 1250
Plugins : mimeTools.dll NppConverter.dll NppExport.dll

**Comment by alankilborn** [comment]
Maybe the menu choice should say it works on unsigned integers.
If you use the "Decimals" menu choice(s), it works.

**Comment by WinterSilence** [comment]
@alankilborn it's looks like bug

**Comment by lisareichelson** [comment]
Hi, can I work on this?

**Comment by alankilborn** [comment]
Maybe it could be the user manual that needs work here.  See [HERE](https://npp-user-manual.org/docs/editing/#:~:text=As%20Integers%20means%20that%2010%20will%20sort%20as,9.876%20as%20decimal%20numbers%20and%20sort%20them%20numerically).  @pryrt what do you think?

Or possibly that plus a menu text change, as I suggested before.

**Comment by ArkadiuszMichalski** [comment]
If the name is `integer`, it should also sort negative ones in the correct order. 

If the name will be [`natural number`](https://en.wikipedia.org/wiki/Natural_number) or  `counting numbers` or ` non-negative integers` or `unsigned integers` then this command should treat negative as positive, or omit such values and not change their order.

Currently it works as described on first post. Although, maybe that's the intended behavior? It looks like it sorts negative and positive separately, with a 0 in between, but negative in fact are treated as positive.

```
init      integers ascendings
-7        -3
-3        -5
-5        -7
1          0
0          1
4          2
2          4
```

**Comment by pryrt** [comment]
Personally, I don't see the utility of an "integer sort" that sorts in the order -3/-5/-7/0/1/2/4.  I would say it's a bug in the implementation.  If the developers decide not to make a change, then I will update the documentation to match that decision, but I would assume it's a bug that should be fixed, rather than documented around.

**Comment by alankilborn** [comment]
I'm sure it was never meant to sort negative integers, from the time of first implementation.  Why? It's too obvious of a miss.
IMO, change its name (to reflect unsigned) or remove the sort entirely (as one of the others does it correctly anyway).

**Comment by alankilborn** [comment]
>then this command should treat negative as positive, or omit such values and not change their order.

I don't see much logic in that.  It should be a precondition of the sort that you only have the correct type of values over the entire sorting range.  If you don't, don't complain about unexpected behavior.

**Comment by ArkadiuszMichalski** [comment]
@donho This looks like regression introduced by https://github.com/notepad-plus-plus/notepad-plus-plus/pull/4413 in [version 7.5.7](https://github.com/notepad-plus-plus/notepad-plus-plus/wiki/Changes#757). The previous versions work fine when we taken into account only numbers. Reading the [issue](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/2025), I don't see that the author's intention was to completley omit negative numbers.

If now it should be only for natural numbers, then this name should be corrected, and it is worth adding in the documentation how negative values are treated.

**Comment by sherwinnie** [comment]
Hi, this is a student from University of Michigan. We found out this issue and it seems like the regression bug introduced by #4413 was made by a previous student from University of Michigan as a school project. We think we will be able to fix this bug and redeem the mistake. Is it okay if we claim this issue?

**Comment by sherwinnie** [comment]
We have implemented a sorting algorithm that takes negative numbers into account. Essentially, it behaves like natural sort, but it also takes in dashes (-) for negative numbers.

However, there are an edge case that we want to ask the community what the intended behavior should be.

Suppose we want to compare "-01" and "-1". Numerically, they have the same value. Then, which one should take priority? Should it be sorted ascendingly to:
```
-01
-1
```

or 

```
-1
-01
```

We think the second option makes more sense, since the numerical value tied, we should compare them as two strings, in which case, the longer string has higher priority.

**Comment by rdipardo** [comment]
I'm pretty sure [ASCII character order][0] is how you decide those things:

 ```
 -01
 -1
 ```

Any rational file system would do the same:

~~~sh-session
$ tree -s
.
├── [          0]  -01.txt
└── [          0]  -1.txt
~~~

[0]: https://en.wikipedia.org/wiki/ASCII#Character_order

**Comment by sherwinnie** [comment]
> I'm pretty sure [ASCII character order](https://en.wikipedia.org/wiki/ASCII#Character_order) is how you decide those things:
> 
> ```
> -01
> -1
> ```
> 
> Any rational file system would do the same:
> 
> ```
> $ tree --sort=name
> .
> ├── -01.txt
> └── -1.txt
> ```

I see. Thank you!

**Comment by donho** [comment]
Tested  PR #11481, it solved also issue #2025:
```
2 3
1 4
2 -10
-2 20
```
